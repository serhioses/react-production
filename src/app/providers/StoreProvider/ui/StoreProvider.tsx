import { PropsWithChildren } from 'react';
import { Provider as ReduxProvider } from 'react-redux';

import { createStore } from 'app/providers/StoreProvider/config/store';

const store = createStore();

export function StoreProvider({ children }: PropsWithChildren) {
  return <ReduxProvider store={store}>{children}</ReduxProvider>;
}
