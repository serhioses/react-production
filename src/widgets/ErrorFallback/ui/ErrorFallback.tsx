import { FallbackProps } from 'react-error-boundary';
import { useTranslation } from 'react-i18next';

import { Alert, AlertDescription, AlertIcon, AlertTitle } from '@chakra-ui/react';

export function ErrorFallback({ error }: FallbackProps) {
  const { t } = useTranslation();

  return (
    <Alert
      status="error"
      sx={{ flexDirection: 'column', alignItems: 'center', justifyContent: 'center' }}
    >
      <AlertIcon />
      <AlertTitle>{t('errorFallbackTitle')}</AlertTitle>
      <AlertDescription>{error.message}</AlertDescription>
    </Alert>
  );
}
