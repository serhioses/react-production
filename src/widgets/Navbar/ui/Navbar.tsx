import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { Link as RouterLink } from 'react-router-dom';

import { MoonIcon, SunIcon } from '@chakra-ui/icons';
import {
  Box,
  IconButton,
  Flex,
  Link,
  SystemStyleObject,
  useColorMode,
  Button,
  HStack,
  StackDivider,
} from '@chakra-ui/react';

import { ERoutes } from 'shared/enums/routes';
import { useLazyModal } from 'shared/hooks/useLazyModal';

import { selectUserAuthData, userActions } from 'entities/User/public-api';

import { LoginModalLazy } from 'features/AuthByUsername/public-api';

import { useAppDispatch } from 'app/providers/StoreProvider/config/hooks';

type TNavbarProps = {
  sx?: SystemStyleObject;
};

export function Navbar({ sx }: TNavbarProps) {
  const { toggleColorMode, colorMode } = useColorMode();
  const { t, i18n } = useTranslation();
  const { ModalComponent: LoginModal, isOpen, onOpen, onClose } = useLazyModal(LoginModalLazy);
  const authData = useSelector(selectUserAuthData);
  const dispatch = useAppDispatch();

  function changeLang(lng: string): void {
    if (lng !== i18n.language) {
      i18n.changeLanguage(lng);
    }
  }

  function logOut(): void {
    dispatch(userActions.resetAuthData());
  }

  return (
    <Box
      sx={{
        display: 'flex',
        alignItems: 'center',
        width: '100%',
        height: 50,
        padding: 5,
        backgroundColor: colorMode === 'light' ? 'gray.500' : 'white',
        color: colorMode === 'light' ? 'white' : 'black',
        boxShadow: 'md',
        ...sx,
      }}
      data-testid="navbar"
    >
      <Flex sx={{ marginInline: 'auto 24px', gap: 4 }}>
        <Link as={RouterLink} to={ERoutes.Main}>
          {t('home')}
        </Link>
        <Link as={RouterLink} to={ERoutes.About}>
          {t('about')}
        </Link>
      </Flex>
      <IconButton
        variant="ghost"
        colorScheme={colorMode === 'light' ? 'gray' : ''}
        onClick={toggleColorMode}
        icon={colorMode === 'light' ? <MoonIcon /> : <SunIcon />}
        aria-label={t('toggleTheme')}
      />
      <HStack divider={<StackDivider />} marginInlineEnd={8}>
        <Button
          size="sm"
          variant="link"
          color="inherit"
          sx={{ opacity: i18n.language !== 'ru' && i18n.language !== 'ru-RU' ? 1 : 0.75 }}
          onClick={() => changeLang('en')}
        >
          EN
        </Button>
        <Button
          size="sm"
          variant="link"
          color="inherit"
          sx={{ opacity: i18n.language === 'ru' || i18n.language === 'ru-RU' ? 1 : 0.75 }}
          onClick={() => changeLang('ru')}
        >
          RU
        </Button>
      </HStack>
      {authData ? (
        <Button
          onClick={logOut}
          variant="link"
          color="inherit"
          sx={{ textDecoration: 'underline' }}
        >
          {t('logout')}
        </Button>
      ) : (
        <>
          <Button
            onClick={onOpen}
            variant="link"
            color="inherit"
            sx={{ textDecoration: 'underline' }}
          >
            {t('login')}
          </Button>
          <LoginModal isOpen={isOpen} onClose={onClose} />
        </>
      )}
    </Box>
  );
}
