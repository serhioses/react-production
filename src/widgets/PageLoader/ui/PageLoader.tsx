import { Center, Spinner, SpinnerProps } from '@chakra-ui/react';

export function PageLoader(props: SpinnerProps) {
  return (
    <Center sx={{ position: 'absolute', inset: 0 }}>
      <Spinner size="xl" {...props} />
    </Center>
  );
}
