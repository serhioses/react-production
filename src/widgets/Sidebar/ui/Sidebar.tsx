import { useState } from 'react';

import { Box, Button } from '@chakra-ui/react';

import GradeIcon from 'shared/assets/icons/grade.svg';

export function Sidebar() {
  const [collapsed, setCollapsed] = useState(false);

  function toggleCollapsed(): void {
    setCollapsed((c) => !c);
  }

  return (
    <Box
      sx={{
        width: collapsed ? '80px' : '300px',
        height: 'calc(100dvh - 50px)',
        background: 'black',
        position: 'relative',
        transition: '0.1s ease-in',
      }}
      data-testid="sidebar"
    >
      <Button onClick={toggleCollapsed} data-testid="sidebar-toggle">
        <GradeIcon />
      </Button>
    </Box>
  );
}
