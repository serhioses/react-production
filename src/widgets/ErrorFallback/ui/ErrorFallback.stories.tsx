import type { Meta, StoryObj } from '@storybook/react';

import { ErrorFallback } from 'widgets/ErrorFallback/ui/ErrorFallback';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: 'Widgets/ErrorFallback',
  component: ErrorFallback,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
} satisfies Meta<typeof ErrorFallback>;

export default meta;

type Story = StoryObj<typeof meta>;

const Template: Story = {
  render: (args) => {
    return <ErrorFallback {...args} />;
  },
  args: {
    error: { message: 'msg' },
    resetErrorBoundary: () => null,
  },
};

export const Basic = {
  ...Template,
};
