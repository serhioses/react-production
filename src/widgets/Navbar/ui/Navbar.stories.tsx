import type { Meta, StoryObj } from '@storybook/react';

import { Navbar } from 'widgets/Navbar/ui/Navbar';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export

const meta = {
  title: 'Widgets/Navbar',
  component: Navbar,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
} satisfies Meta<typeof Navbar>;

export default meta;

type Story = StoryObj<typeof meta>;

const Template: Story = {
  render: (args) => {
    return <Navbar {...args} />;
  },
};

export const Basic = {
  ...Template,
};
