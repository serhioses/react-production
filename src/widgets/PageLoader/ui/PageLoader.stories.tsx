import type { Meta, StoryObj } from '@storybook/react';

import { PageLoader } from 'widgets/PageLoader/ui/PageLoader';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: 'Widgets/PageLoader',
  component: PageLoader,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
  argTypes: {
    color: { control: 'text' },
    emptyColor: { control: 'text' },
    label: { control: 'text' },
    size: {
      options: ['xs', 'sm', 'md', 'lg', 'xl'],
      control: 'select',
    },
    speed: { control: 'text' },
    thickness: { control: 'text' },
    variant: { control: 'text' },
  },
  args: {
    size: 'sm',
  },
} satisfies Meta<typeof PageLoader>;

export default meta;

type Story = StoryObj<typeof meta>;

const Template: Story = {
  render: (args) => {
    return <PageLoader {...args} />;
  },
};

export const Basic = {
  ...Template,
};
