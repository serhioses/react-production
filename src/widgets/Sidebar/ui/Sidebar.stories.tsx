import type { Meta, StoryObj } from '@storybook/react';

import { Sidebar } from 'widgets/Sidebar/ui/Sidebar';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export

const meta = {
  title: 'Widgets/Sidebar',
  component: Sidebar,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
} satisfies Meta<typeof Sidebar>;

export default meta;

type Story = StoryObj<typeof meta>;

const Template: Story = {
  render: (args) => {
    return <Sidebar {...args} />;
  },
};

export const Basic = {
  ...Template,
};
