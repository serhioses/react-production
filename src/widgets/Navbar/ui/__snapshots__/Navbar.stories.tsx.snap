// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/Navbar Basic smoke-test 1`] = `
<div data-testid="navbar"
     class="css-1dkag26"
>
  <div class="css-1lzq4gc">
    <a class="chakra-link css-spn4bz"
       href="/"
    >
      Home
    </a>
    <a class="chakra-link css-spn4bz"
       href="/about"
    >
      About
    </a>
  </div>
  <button type="button"
          class="chakra-button css-15bt6gt"
          aria-label="Toggle theme"
  >
    <svg viewbox="0 0 24 24"
         focusable="false"
         class="chakra-icon css-onkibi"
         aria-hidden="true"
    >
      <path fill="currentColor"
            d="M21.4,13.7C20.6,13.9,19.8,14,19,14c-5,0-9-4-9-9c0-0.8,0.1-1.6,0.3-2.4c0.1-0.3,0-0.7-0.3-1 c-0.3-0.3-0.6-0.4-1-0.3C4.3,2.7,1,7.1,1,12c0,6.1,4.9,11,11,11c4.9,0,9.3-3.3,10.6-8.1c0.1-0.3,0-0.7-0.3-1 C22.1,13.7,21.7,13.6,21.4,13.7z"
      >
      </path>
    </svg>
  </button>
  <div class="chakra-stack css-1p1m4ay">
    <button type="button"
            class="chakra-button css-gqe90k"
    >
      EN
    </button>
    <div class="chakra-stack__divider css-14rtrcu">
    </div>
    <button type="button"
            class="chakra-button css-vklvw0"
    >
      RU
    </button>
  </div>
</div>
<span id="__chakra_env"
      hidden
>
</span>
`;
